### ISSUE TITLE
FPP remote not showing LED output

### ISSUE THREAD

**Issue by intermittech** [forkineye/ESPixelStick#472]
**ESPixelStick Firmware Version**
4.0-Beta4 (release download from Github)

**Hardware Version**
QuinLED-Dig-Uno with prototype AE+ hat which has MicroSD card slot)

**Binary release or compiled yourself?**
4.0-Beta4 (release download from Github)

**Operating System (and version)**
Windows 10 and 11

**Web Browser (and version)**
Chrome

**Access Point**
Mikrotik wAP-AC

**SDcard pinout for AE+ hat**
SD MISO pin 19
SD Clock pin 18
SD MOSI pin 23
SD Chip Select pin 5

**LED configuration**
Output 1 is enabled at GPIO16 with 300 Pixel Count

**Describe the bug**
Running the latest ESPixelStick 4.0-beta4 on a custom ESP32 setup (QuinLED-Dig-Uno with prototype AE+ hat which has MicroSD card slot).

E1.31 mode works fine, uploading .fseq using FPP connect or web interface works fine. Even local playback of the .fseq file works great.

But when using xSchedule to start the same .fseq file for remote playback, it seems to start and all signs indicate everything is working fine but there is no output to the LEDs or in Diagnostics.

p.s. I can attach a .fseq file to this git issue but it's literally the most simple thing you can make in xLights and as mentioned, local playback of the file works perfectly.

**Comment by SmAsH-jl** [comment]
so far. v4beta works to play fseq locally. does not work but does show its playing the fseq remotely through FPP master,

 none of the latest ci builds read the sd card.
[FileMgr] Configuration File: '/config.json' saved 348 bytes.
[FileMgr] Configuration File: '/config.json' loaded.
[FileMgr] Configuration File: '/input_config.json' saved 949 bytes.
[FileMgr] Configuration File: '/output_config.json' saved 4358 bytes.
[ 73019][E][sd_diskio.cpp:126] sdSelectCard(): Select Failed
[ 73019][E][sd_diskio.cpp:790] sdcard_mount(): f_mount failed: (3) The physical drive cannot work
[ 73322][E][sd_diskio.cpp:126] sdSelectCard(): Select Failed
[FileMgr] No SD card installed
[FileMgr] Configuration File: '/input_config.json' loaded.


but if i leave the exact same setup plugged into USB and then reflash with v4beta. it reads it no problem

i do have to go change the chip select pin on v4b to pin 5, but thats fine then filemanger section pops right up.

also any of the ci builds i have tested have weird default sd pinouts set  13, 14, 15, and 16.. instead of v4b having  19, 23, 18, and 4, by default for the gpio's

**Comment by MartinMueller2003** [comment]
The GPIOS for each of the supported platforms were chosen specifically for that platform. You can change the GPIOs being used vi the browser or you can request support for a new platform.

In the end it looks like that statement is: FPP remote when using an FPP master to control the device is not outputting data to the lights. FYI: The diagnostics output is a separate output from the lights so if the Diagnostics screen is dark then there is no data for the pixel output stage to send either.

**Comment by MartinMueller2003** [comment]
I am playing a test sequence controlled by my FPP with no issues. Please send me your file and I will see if there is something odd about it.

**Comment by SmAsH-jl** [comment]
Hi Martin, Further testing, I dropped down to version 4beta3 under the release candidates section and it works flawlessly, 

the difference I notice there is that the fpp remote was moved from primary input to secondary input. 

Here is the test sequence. just a one output string of 20 pixels.  this works in beta 3 but not in beta 4, remotely from fpp master. 
[test.zip](https://github.com/forkineye/ESPixelStick/files/8054392/test.zip)

**Comment by TechTurtleTube** [comment]
Using beta b3 : works perfectly 

Using beta b4:
 I can't get output to lights using master from xschedule  even though file it says it's getting packets but playing local plays and from xlights works 


Ci build Using remote from xschedule:

 I can also confirm that e1.31 seems smoother than playing locally with master sync also notice a stall and delay when The seq ends and the lights go out using sd

Basically freezing last Frame till it gets some signal seems to have alot of sync issues 

During testing I have used fpp connect 
Uploaded outputs
Single line on port one 

Xschedule modes : unicast ,broadcast, or multicast

E1.31 disabled as primary

**Comment by MartinMueller2003** [comment]
Would someone please try the images created by PR 469. I suspect some of these issues have been resolved.

**Comment by intermittech** [comment]
> Would someone please try the images created by PR 469. I suspect some of these issues have been resolved.

As far as I can see this is the newest Ci, just to make sure I tried again and unlike Beta4 it doe give FPP remote output but both playing locally or remote FPP it's very choppy and lots of sync events. 

Total Sync Packets: | 366
Sync Adjustments: | 239

On Beta3 everything was much smoother (same .fseq file) and it started with a few sync adjustments and then only the sync packets counter goes up and it just stays in sync.

Are you sure the build files where updated? Because I believe these are the same files that where there for 7 days already? Sadly I can't get it to compile myself, I'll try again and see if that makes a difference. :)

**Comment by MartinMueller2003** [comment]
crashes? do you have a backtrace for the crash? were you accessing the ui at the time?

**Comment by brmeadows2841** [comment]
I am running the CLI version and am getting this error then a reboot.

ESP32 CAM

 

ESPixelStick_Firmware-4.0-ci1794899715

 

 

 

 

[FileMgr] Configuration File: '/config.json' loaded.

Guru Meditation Error: Core  0 panic'ed (LoadProhibited). Exception was unhandled.

 

Core  0 register dump:

PC      : 0x40093313  PS      : 0x00060933  A0      : 0x80093a35  A1      : 0x3ffbd0a0  

A2      : 0x3f800014  A3      : 0x3f800014  A4      : 0x80000000  A5      : 0x00060723  

A6      : 0x00060720  A7      : 0x00000001  A8      : 0x3f8006c4  A9      : 0x94d5556c  

A10     : 0x3f8006cc  A11     : 0x00000002  A12     : 0x3f80001c  A13     : 0x55554ea4  

A14     : 0xfffffffc  A15     : 0x55555554  SAR     : 0x0000001c  EXCCAUSE: 0x0000001c  

EXCVADDR: 0x94d55570  LBEG    : 0x4008ba70  LEND    : 0x4008ba7b  LCOUNT  : 0x00000000  

 

 

Backtrace:0x40093310:0x3ffbd0a00x40093a32:0x3ffbd0c0 0x40093c84:0x3ffbd0e0 0x40084756:0x3ffbd100 0x400847d2:0x3ffbd130 0x4008446d:0x3ffbd180 0x40191b95:0x3ffbd1a0 0x40161328:0x3ffbd1d0 0x40161539:0x3ffbd1f0 0x4016552e:0x3ffbd210 0x40161adb:0x3ffbd230 0x40151b62:0x3ffbd250 0x40151c29:0x3ffbd270 0x40150364:0x3ffbd290 0x4018fd89:0x3ffbd2b0 

 

 

 

 

ELF file SHA256: 0000000000000000

 

Rebooting...

ets Jun  8 2016 00:22:57

**Comment by intermittech** [comment]
> crashes? do you have a backtrace for the crash? were you accessing the ui at the time?

Did some more testing but yes this happens mainly when accessing the GUI and configuration screens, I'll try and see if I can capture some debug data but since you said the current Ci build has lots of bugs anyway I'm not sure that's going to do a lot of good. I can't seem to find the build you are talking about that should resolve a lot of these issues as you mentioned, for me there is basically no difference between latest Ci, Master or the change you mentioned.

Beta3 seems to pretty much do everything correctly where Beta4 is slow, most often doesn't do FPP output, sync issues and crashes. If you say current Master self built should have fixes for that I'll try that again just to make sure and then also capture some debug data! :)

**Comment by MartinMueller2003** [comment]
The latest PR fixed that "slow" issue. Most of the crashes are due to low memory issues on ESP8266. ESP32 has more RAM so it has fewer issues.

**Comment by MartinMueller2003** [comment]
As for freeze on the last frame, The FSEQ player will reach the last frame, output it and go to sleep. The lights will stay that way until a new file starts playing, a clear buffer command is received from the FPP or the blank timer expires.

**Comment by brmeadows2841** [comment]
> > Would someone please try the images created by PR 469. I suspect some of these issues have been resolved.
> 
> As far as I can see this is the newest Ci, just to make sure I tried again and unlike Beta4 it doe give FPP remote output but both playing locally or remote FPP it's very choppy and lots of sync events.
> 
> Total Sync Packets: | 366 Sync Adjustments: | 239
> 
> On Beta3 everything was much smoother (same .fseq file) and it started with a few sync adjustments and then only the sync packets counter goes up and it just stays in sync.
> 
> Are you sure the build files where updated? Because I believe these are the same files that where there for 7 days already? Sadly I can't get it to compile myself, I'll try again and see if that makes a difference. :)
> 
> --update I was able to compile the newest Master and flash it, but I see no difference vs the latest Ci auto build myself. Remote FPP (and local play) works sometimes (it seems) but it's really choppy and it also crashes quickly.

I'm not trying to bomb this issue but I did notice if xlight's output to light is turned on, the ESPixelstick device, EPS32CAM in my case,  will stop playing on diagnostic screen even if a seq is not playing from xlights, there is several seconds delay. 
I suspect this is by design, WLED has the same behavior.

**Comment by MartinMueller2003** [comment]
I am seeing an odd issue as I poke at this. My code runs fine on ESP8266, ESP32 LoLin PRO, ESP32 TTGO T8. I get SD file read errors on the ESP32-CAM board. There is something about the ESP32-CAM SD interface.

**Comment by MartinMueller2003** [comment]
So every time the WiFi changes state I get errors on the SD interface (CAM card only). If I set the WiFi sleep mode then the SD card interface dies entirely. The Status shows playing but since no data is coming from the SD card there are no lights. Again, this is on the CAM card only. My other cards are working fine.

**Comment by MartinMueller2003** [comment]
Tried going to the latest Framework and the WiFi issues got worse. Admittedly this is a pre-beta Use at your own risk version so some of the issues should get resolved before the framework gets released.

**Comment by MartinMueller2003** [comment]
ok. Found a version of the framework that does not generate errors reading the SD card when the WiFi changes state. You cannot build this using the Arduino IDE. You MUST build using platformio in vscode. Code has been pushed to my branch and a pull request is open.

**Comment by MartinMueller2003** [comment]
Just sent a request to the QuinLed team to help me get information needed to create a pre-compiled image for the QuinnLed ESP32 + DigiQuad board. Once I get the data, I will add the board to our lists. In the meantime, my branch has some improvements in performance and has a lot of bug fixes from the Framework IDF teams.

**Comment by MartinMueller2003** [comment]
Changes are in main
